packages/mui-base/src/useNumberInput/useNumberInput.ts

```diff
@@ -0,0 +1,371 @@
+      'onBlur',
+      'onChange',
+      'onFocus',
+      'onValueChange',
+    ]);
+
+    const externalEventHandlers = { ...propsEventHandlers, ...extractEventHandlers(externalProps) };
+
+    return {
+      ...externalProps,
+      ...externalEventHandlers,
+      onClick: handleClick(externalEventHandlers),
+    };
+  };
+
+  const getInputProps = <TOther extends Record<string, any> = {}>(
+    externalProps: TOther = {} as TOther,
+  ): UseNumberInputInputSlotProps<TOther> => {
+    const propsEventHandlers: Record<string, React.EventHandler<any> | undefined> = {
+      onBlur,
+      onChange,
+      onFocus,
+    };
+
+    const externalEventHandlers = { ...propsEventHandlers, ...extractEventHandlers(externalProps) };
+
+    const mergedEventHandlers = {
+      ...externalProps,
+      ...externalEventHandlers,
+      onFocus: handleFocus(externalEventHandlers),
+      onChange: handleInputChange(externalEventHandlers),
+      onBlur: handleBlur(externalEventHandlers),
+      onKeyDown: handleKeyDown(externalEventHandlers),
+    };
+
+    const displayValue = (focused ? dirtyValue : value) ?? '';
+
+    return {
+      ...mergedEventHandlers,
+      type: 'text',
+      // TODO: check to see if SR support is still weird
+      role: 'spinbutton',
+      'aria-invalid': errorProp || undefined,
+      defaultValue: defaultValueProp as number | undefined,
+      ref: handleInputRef,
+      value: displayValue as number | undefined,
+      'aria-valuenow': displayValue as number | undefined,
+      'aria-valuetext': String(displayValue),
+      'aria-valuemin': min,
+      'aria-valuemax': max,
+      autoComplete: 'off',
+      autoCorrect: 'off',
+      required: requiredProp,
+      'aria-disabled': disabledProp,
+      disabled: disabledProp,
+    };
+  };
+
+  const isIncrementDisabled = isNumber(value) ? value >= (max ?? Number.MAX_SAFE_INTEGER) : false;
+
+  const getIncrementButtonProps = <TOther extends Record<string, any> = {}>(
+    externalProps: TOther = {} as TOther,
+  ): UseNumberInputIncrementButtonSlotProps<TOther> => {
+    return {
+      ...externalProps,
+      // the button should be tab-able if the input is readonly
+      tabIndex: -1,
+      disabled: isIncrementDisabled,
+      'aria-disabled': isIncrementDisabled,
+      onClick: handleStep('up'),
+    };
+  };
+
+  const isDecrementDisabled = isNumber(value) ? value <= (min ?? Number.MIN_SAFE_INTEGER) : false;
+
+  const getDecrementButtonProps = <TOther extends Record<string, any> = {}>(
+    externalProps: TOther = {} as TOther,
+  ): UseNumberInputDecrementButtonSlotProps<TOther> => {
+    return {
+      ...externalProps,
+      // the button should be tab-able if the input is readonly
+      tabIndex: -1,
+      disabled: isDecrementDisabled,
+      'aria-disabled': isDecrementDisabled,
+      onClick: handleStep('down'),
+    };
+  };
+
+  return {
+    disabled: disabledProp,
+    error: errorProp,
```

Comment:
I think we may omit the values that are just passed from the input props and not calculated or processed in anyway. I know that `useButton` for example returns `disabled`, but I'm leaning towards removing it. It will reduce the confusion of the users with regards to which value to use as the source of truth.

Reply:
This makes sense – I'll keep the ones that `useInput` has for now so they are consistent